server/src/com/thoughtworks/go/config/update/DeleteTemplateConfigCommand.java

```diff
@@ -54,7 +55,15 @@ public boolean isValid(CruiseConfig preprocessedConfig) {
 
     @Override
     public boolean canContinue(CruiseConfig cruiseConfig) {
-        return super.canContinue(cruiseConfig) && doesTemplateExist(cruiseConfig);
+        return isUserAuthorized() && doesTemplateExist(cruiseConfig);
```

Comment:
The check should be inverted else it will throw a null pointer exception. The same for Update-template-config-command